gcc/rust/backend/rust-compile-intrinsic.cc

```diff
@@ -15,10 +15,29 @@
 // <http://www.gnu.org/licenses/>.
 
 #include "rust-compile-intrinsic.h"
+#include "rust-compile-type.h"
+#include "rust-compile-fnparam.h"
+#include "rust-tree.h"
 
 namespace Rust {
 namespace Compile {
 
+static tree
+offset_intrinsic_handler (Context *ctx, TyTy::BaseType *fntype);
+
+typedef tree (*generic_intrinsic_handler) (Context *ctx,
+					   TyTy::BaseType *fntype);
+struct BuiltinGenericIntrinsic
+{
+  std::string identifier;
+  generic_intrinsic_handler handler;
+};
+static const BuiltinGenericIntrinsic generic_intrinsics[]
+  = {{"offset", &offset_intrinsic_handler}};
+
+#define GENERIC_INTRINSIC_SIZE                                                 \
+  sizeof (generic_intrinsics) / sizeof (BuiltinGenericIntrinsic)
+
```

Comment:
I think you can make this a little more "Cplusplussy"

```suggestion
static tree
offset_intrinsic_handler (Context *ctx, TyTy::BaseType *fntype);

static const std::map<std::string, std::function<tree (Context *, TyTy::BaseType *)>> generic_intrinsics[]
  = {{"offset", offset_intrinsic_handler}};
```
Same exact semantics for defining them but shorter + you get faster lookup + a `.size()` method to avoid the macro

Reply:
fixed